### DISCUSSION TITLE
Getting older data - is there a usage limit?

### DISCUSSION THREAD

**Discussion by vacchiano** [dpguthrie/yahooquery#138] [General]
I have a few hundred tickers I run through grabbing the financial_data and key_stats from Ticker.

When I call them in 1-off situations the key_stats data + dates returned are the most recent.

When I loop through my list of tickers and call the above, sometimes the data & dates are old.

For ex I'll get a qtr date of 6/29/22 instead of 9/29/22 (which I get in a 1-off call).

Wasn't sure if some sort of limit that's causing this or any other ideas.

**Comment by EnigmaNZ** [comment]
I've had this issue as well. I have noted what is updating correctly and which is old data with the functions that call.
(Also these functions were called in this order)

Within financial_data:
- currentPrice = Up to Date
- totalCash = Up to Date
- totalDebt = Up to Date

Within index_trend:
- +5y growth estimates = Out of Date

Within key_stats:
- priceToBook = Out of Date
- enterpriseValue = Out of Date

Within summary_detail:
- trailingPE = Out of Date
- priceToSalesTrailing12Months = Out of Date
- marketCap = Out of Date

Within cash_flow:
- FreeCashFlow = Out of Date

Any help related to updated table layouts or with vacchiano's comment is there a limit to data being pulled?

**Reply by vacchiano** [reply]
Thanks for the detail. I haven't checked too in depth each data point but just waiting a short time (~1sec) between calls helped me

**Reply by EnigmaNZ** [reply]
Awesome. I just noticed more info was being followed up on in the issues so have posted in there too
https://github.com/dpguthrie/yahooquery/issues/151

I'll see if that fixes my issue too
